### PR TITLE
first mock

### PR DESCRIPTION
**Pull request by martinkalaydzhiev-taulia**
@lenikirilov-taulia

### CODE REVIEW

File: src/main/java/EmailService.java [hunk 1]
```diff
@@ -0,0 +1,3 @@
+public interface EmailService {
+    void sentEmail();
```
Comment (lenikirilov-taulia) [inline src/main/java/EmailService.java]: should be present tense - "sendEmail()"

File: src/test/java/SimpleMockSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        def emailService = Mock(EmailService)
+        def flights = new Flights([FlightGenerator.generateFlight(["destination": Airports.PARIS]),
+                                   FlightGenerator.generateFlight(["destination": Airports.PARIS])], emailService)
+
+        when:
+        FlightsReporter.getFlightsToAndSendEmail(flights, Airports.PARIS, emailService)
+
+        then:
+        1 * emailService.sentEmail()
+    }
```
Comment (lenikirilov-taulia) [inline src/test/java/SimpleMockSpec.groovy]: No, this is not what I wanted to try out :) The service is part of the normal class and object. It makes not difference that you are calling Flights from a static method. This is the same as the above

what I wanted you to do is to call directly emailService from the static method. Add it as a field to the FlightReporter and do another test.

File: src/test/java/SimpleMockSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        given:
+        def emailService = Mock(EmailService)
+        def flights = new Flights([FlightGenerator.generateFlight(["destination": Airports.PARIS]),
+                                   FlightGenerator.generateFlight(["destination": Airports.PARIS])], emailService)
+
+        when:
+        FlightsReporter.getAllFlightsToAndSendEmail(flights, Airports.PARIS)
+
+        then:
+        1 * emailService.sentEmail() >> println("ok")
```
Comment (lenikirilov-taulia) [inline src/test/java/SimpleMockSpec.groovy]: 👍

File: src/test/java/SimpleMockSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class SimpleMockSpec extends Specification {
+    @Unroll
+    def "testing email service mock from class Flights"() {
```
Comment (lenikirilov-taulia) [inline src/test/java/SimpleMockSpec.groovy:6]: Now add a test where you have zero flights and you expect the sendEmail() to not be called.

Also a 3rd test is to expect an issue with the 3rd party, and that it would throw an exception (for example "AccessDeniedException"

File: src/main/java/FlightsReporter.groovy [hunk 5]
```diff
@@ -4,11 +4,19 @@ import java.time.LocalDateTime
 import java.time.temporal.ChronoUnit
 
 class FlightsReporter {
+    //EmailService emailService
```
Comment (lenikirilov-taulia) [inline src/main/java/FlightsReporter.groovy]: This is the 2nd thing I want you to try

File: src/main/java/Flights.groovy [hunk 6]
```diff
@@ -3,17 +3,33 @@ import java.time.temporal.ChronoUnit
...
     }
-
     Flights(List<Flight> flights) {
         this.flights = flights
     }
 
+    Flights(List<Flight> flights, EmailService emailService) {
+        this.flights = flights
+        this.emailService = emailService
+    }
```
Comment (lenikirilov-taulia) [inline src/main/java/Flights.groovy:19]: OK cool. This example is fine. Although from OOP point of view, Flights shouldn't have any external dependencies, and the email part must be part of the reporter class. But let's leave it so that we compare the 2

**Review by lenikirilov-taulia — CHANGES_REQUESTED** [review]
Good for the 1st part of the excercise - mocking a field
Now add a static field for the 2nd part

**Review by lenikirilov-taulia — APPROVED** [review]
Old left behind PR :) 

I am expecting a new one for spring
